packages/sdk/src/funding/index.ts

```diff
@@ -64,31 +60,22 @@ export class FundingSDK {
   }
 
   async getFlipBalance(): Promise<bigint> {
-    const flipAddress = getTokenContractAddress('FLIP', this.options.network);
-    const flip = ERC20__factory.connect(flipAddress, this.options.signer);
-    const balance = await flip.balanceOf(
-      await this.options.signer.getAddress(),
-    );
-    return balance.toBigInt();
+    return getFlipBalance(this.options.network, this.options.signer);
   }
 
   /**
    * @param amount the amount of FLIP to request approval for
    * @returns the transaction hash or null if no approval was required
    */
-  async requestFlipApproval(
+  async approveStateChainGateway(
```

Comment:
how do we determine if we need an approval in the auctions app? should we expose a method like `needsStateChainGatewayApproval`?